pkg/operator/ceph/cluster/cluster.go

```diff
@@ -72,7 +72,8 @@ func newCluster(c *cephv1.CephCluster, context *clusterd.Context) *cluster {
 		Spec:      &c.Spec,
 		context:   context,
 		stopCh:    make(chan struct{}),
-		ownerRef:  ClusterOwnerRef(c.Namespace, string(c.UID)),
+		ownerRef:  ownerRef,
+		mons:      mon.New(context, c.Namespace, c.Spec.DataDirHostPath, c.Spec.Network.HostNetwork, ownerRef),
```

Comment:
Is there a way we can start reducing the number of parameters we are passing around? I feel like I spend a lot of time changing function definitions and calls. Why don't we choose to pass around the cluster spec more often so the daemon operator code paths could have access to all the information they might need?

Reply:
It's a great question. I'll look at this again and consider just passing the ClusterSpec type and reduce the params needed. It's a fine line between making it clear what is expected in a function call and passing structs with more properties than are needed.

Reply:
Ok, pushed changes to use the cluster spec instead of individual args. While it does simplify the code, I'm still torn between passing minimal info that a package needs vs an object which contains a lot of extra properties.

Reply:
I share some of your hesitation. Maybe a middle ground (for the future) would be to make a struct that contains global cluster config that any daemon might want to know (e.g., cluster context, rook and ceph versions, hostPath, namespace, owner ref to cluster object, etc.), and then the individual daemons would get that global spec as well as the spec specific to them (like filesystem spec for file/mdses, storage for osds).